File: services/api/src/restaurant_os_api/modules/restaurant/domain/entities/modifier_group.py
"""ModifierGroup entity.

Restaurant Platform Architecture SS3.1: a named set of choices for a
menu item, shareable across multiple items (Data Architecture v2.0
Group F). Name uniqueness is deliberately **not** enforced -- a group
named "Size" legitimately repeats across unrelated item families.
"""

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from enum import StrEnum


class ModifierSelectionType(StrEnum):
    SINGLE = "single"
    MULTIPLE = "multiple"


@dataclass(slots=True)
class ModifierGroup:
    id: str
    tenant_id: str
    name: str
    selection_type: ModifierSelectionType
    created_at: datetime

File: apps/admin-web/src/types/modifier-group.ts
/**
 * Mirrors modules/restaurant/presentation/api/v1/modifier_group_router.py's
 * ModifierGroupResponseSchema / Create·Update ModifierGroupRequestSchema
 * and the domain's ModifierSelectionType StrEnum. ModifierGroup has no
 * FK parent (Data Architecture v2.0 Group F -- it belongs directly to
 * the tenant), unlike every other Restaurant Platform entity built so
 * far. Field names are camelCase on the wire.
 */

export type ModifierSelectionType = "single" | "multiple"

export interface ModifierGroup {
  id: string
  tenantId: string
  name: string
  selectionType: ModifierSelectionType
  createdAt: string
}

export interface CreateModifierGroupRequest {
  name: string
  selectionType: ModifierSelectionType
}

export type UpdateModifierGroupRequest = CreateModifierGroupRequest

export interface ListModifierGroupsParams {
  offset?: number
  limit?: number
}
